include/emitc/emitc_tosa.h

```diff
@@ -137,6 +139,120 @@ Dest fully_connected(Src input, Weights weights, Bias bias) {
   return output;
 }
 
+/// Reduce ops
+// ReduceOp
+template <typename Dest, typename Src, typename Computation>
+inline Dest reduce(Src operand, typename get_element_type<Src>::type initValue,
+                   int64_t dimension, Computation computation) {
+  static_assert(is_tensor<Src>::value, "Expected tensor argument");
+  static_assert(is_tensor<Dest>::value, "Expected tensor result");
+
+  using ET_Src = typename get_element_type<Src>::type;
+  using ET_Dest = typename get_element_type<Src>::type;
```

Comment:
`using ET_Dest = typename get_element_type<Src>::type;`

You're getting the type of `Src` instead of `Dest`.